src/lib.rs

```diff
@@ -412,6 +381,62 @@ impl Watch {
     }
 }
 
+struct WatchEventHandler {
+    watch: Watch,
+    command: Command,
+    child: Child,
+    command_start: Instant,
+}
+
+impl EventHandler for WatchEventHandler {
+    fn handle_event(&mut self, event: Result<Event, notify::Error>) {
+        match event {
+            Ok(event) => {
+                if self
+                    .watch
+                    .compare_event(&event, self.command_start.elapsed())
+                {
+                    log::trace!("Changes detected in {:?}", event);
+                    #[cfg(unix)]
+                    {
+                        let now = Instant::now();
+
+                        unsafe {
+                            log::trace!("Killing watch's command process");
+                            libc::kill(
+                                self.child.id().try_into().expect("cannot get process id"),
+                                libc::SIGTERM,
+                            );
+                        }
+
+                        while now.elapsed().as_secs() < 2 {
+                            std::thread::sleep(Duration::from_millis(200));
+                            if let Ok(Some(_)) = self.child.try_wait() {
+                                break;
+                            }
+                        }
+                    }
+
+                    match self.child.try_wait() {
+                        Ok(Some(_)) => {}
+                        _ => {
+                            let _ = self.child.kill();
+                            let _ = self.child.wait();
+                        }
+                    }
+
+                    log::info!("Re-running command");
+                    self.child = self.command.spawn().expect("cannot spawn command");
+                    self.command_start = Instant::now();
```

Comment:
See here how the "now" variable seems wrong? If it was really a "now" variable you would have used it here instead of calling `Instant::now()`. But you don't because `now` is not exactly now but then time at which you started stopping the process.